src/main/java/teammates/logic/api/Logic.java

```diff
@@ -2622,4 +2622,14 @@ public MimeMessage emailErrorReport(String path, String params, Throwable error)
     @SuppressWarnings("unused")
     private void ____helper_methods________________________________________() {
     }
+    
+    public void extractActiveAndArchivedCourses(List<CourseDetailsBundle> courseBundles, 
```

Comment:
Using parameters as outputs is discouraged. It is a shortcut that works, but it is not common usage (things that can 'surprise' the reader is not recommended) and the popular opinion seems to be against it. If you want to return multiple things, encapsulate them into another class. However, if there are many places this technique can be useful, we can define a convention to make such 'output parameters' more obvious (and hence not 'surprising'). @wkurniawan07 also used this technique.